pkg/queue/health/probe_test.go

```diff
@@ -31,12 +31,12 @@ func TestTcpProbe(t *testing.T) {
 	serverAddr := server.Listener.Addr().String()
 
 	// Connecting to the server should work
-	if TCPProbe(serverAddr, 1*time.Second) != true {
-		t.Error("Expected probe to succeed but it didn't")
+	if err := TCPProbe(serverAddr, 1*time.Second); err != nil {
+		t.Errorf("Expected probe to succeed but it failed with %v", err)
 	}
 
 	// Break the address so probing fails
-	if TCPProbe(serverAddr+"-1", 1*time.Second) != false {
+	if err := TCPProbe(serverAddr+"-1", 1*time.Second); err == nil {
```

Comment:
I think here you're breaking the format of IP address, so net.Dial will fail to parse the address. It's still a valid test, but probably not what you had in mind :-)

Reply:
It's what I actually had in mind, but I think the new version makes a bit more sense.